service-base/src/main/java/org/eclipse/hono/service/VertxBasedHealthCheckServer.java

```diff
@@ -69,13 +70,13 @@
      * @param config The application configuration instance.
      * @throws NullPointerException if vertx is {@code null}.
      * @throws NullPointerException if config is {@code null}.
-     * @throws IllegalArgumentException if health check port is not configured.
      */
     public VertxBasedHealthCheckServer(final Vertx vertx, final ServerConfig config) {
         this.vertx = Objects.requireNonNull(vertx);
         this.config = Objects.requireNonNull(config);
         if (!isSecureEndpointConfigured() && !isInsecureEndpointConfigured()) {
-            throw new IllegalArgumentException("No health check configured");
+            config.setInsecurePort(DEFAULT_HEALTH_CHECK_PORT);
```

Comment:
IMHO we should not overwrite the config properties but instead use `ServerConfig.getPort(String)` when starting the server to fall back to the default port.